webui/react/src/services/decoder.ts

```diff
@@ -444,6 +444,26 @@ const decodeCheckpointWorkload = (data: Sdk.V1CheckpointWorkload): types.Checkpo
   };
 };
 
+/* using any here because this comes from the api as any /*
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+export const decodeMetricStruct = (data: any): Record<string, number> => {
+  const metrics: Record<string, number> = {};
+  Object.entries(data || {}).forEach(([ metric, value ]) => {
+    if (typeof metric === 'string' && (typeof value === 'number' || typeof value === 'string')) {
+      const numberValue = (typeof value === 'number') ? value : parseFloat(value);
```

Comment:
re any: 
nit: I think this could be more robust if have the function take `unknown`and check whether it's an object and go from there. that should be safer because it'll force the implementation to handle other types as well.

```
nesh> Object.entries('aj223' || {})
[
  [ '0', 'a' ],
  [ '1', 'j' ],
  [ '2', '2' ],
  [ '3', '2' ],
  [ '4', '3' ]
]
```

it might be safer to just ignore value === 'string' if we don't expect them to come through as string since parseFloat accepts this 
```
nesh> parseFloat('1.2h34asdf')
1.2
```

Reply:
re: re: any:
i'm happy to do this, but I guess I'm a little confused about the implications:
- the values supplied to this function correspond to either `Metrics.avg_metrics` of items of `Metrics.batch_metrics` as defined in `checkpoint.proto`
- so, based on the protos, `data` will always be a `Struct`/`Record`/_Real Object_™ here.
- so, my assumption would be if you trust the protos, the check is redundant (and potentially confusing for those reading it later)
- and if you don't trust the protos, what can you trust??

In order to make it clear that the assumptions of this function are only valid for the values being supplied here, I would propose putting `decodeMetricStruct` inside of `decodeMetrics`.

Reply:
I guess maybe one concern would be that the protos could change and this code wouldn't know it.

I thought about doing something like this:
```
  const decodeMetricStruct = (data: Sdk.V1Metrics['avgMetrics']): Record<string, number> => {
```
but then if the protos change to a allow a string for that type, it's going to stop your from doing `Object.entries` like you were saying.